packages/wdio-cli/tests/launcher.test.js

```diff
@@ -143,6 +143,17 @@ describe('launcher', () => {
             launcher.endHandler({ cid: '0-5', exitCode: 1, retries: 1, specs: ['a.js'] })
             expect(launcher.schedule).toMatchObject([{ cid: 0, specs: [{ rid: '0-5', files: ['a.js'], retries: 0 }] }])
         })
+
+        it('should requeue retried specfiles correctly', () => {
```

Comment:
"correctly" it's not clear definition. You could create two tests for covering two cases and write specific description